hack/update-vendor.sh

```diff
@@ -356,8 +374,8 @@ xargs -L 100 go mod edit -fmt
 # disallow transitive dependencies on k8s.io/kubernetes
 loopback_deps=()
 kube::util::read-array loopback_deps < <(go mod graph | grep ' k8s.io/kubernetes' || true)
-if [[ -n ${loopback_deps[*]:+"${loopback_deps[*]}"} ]]; then
-  kube::log::error "Disallowed transitive k8s.io/kubernetes dependencies exist via the following imports:" >&22 2>&1
+if (( "${#loopback_deps[@]}" > 0 )); then
```

Comment:
did you make sure this actually fires if there are loopback deps? I can't eyeball this to tell

Reply:
I applied this patch:

```diff
diff --git a/staging/src/k8s.io/sample-apiserver/go.mod b/staging/src/k8s.io/sample-apiserver/go.mod
index 90a3ac47569..a753626d2e7 100644
--- a/staging/src/k8s.io/sample-apiserver/go.mod
+++ b/staging/src/k8s.io/sample-apiserver/go.mod
@@ -102,6 +102,7 @@ require (
 	k8s.io/gengo v0.0.0-20230829151522-9cce18d56c01 // indirect
 	k8s.io/klog/v2 v2.110.1 // indirect
 	k8s.io/kms v0.0.0 // indirect
+	k8s.io/kubernetes v0.0.0 // indirect
 	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.28.0 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
@@ -115,5 +116,6 @@ replace (
 	k8s.io/code-generator => ../code-generator
 	k8s.io/component-base => ../component-base
 	k8s.io/kms => ../kms
+	k8s.io/kubernetes => ../../../..
 	k8s.io/sample-apiserver => ../sample-apiserver
 )
diff --git a/staging/src/k8s.io/sample-apiserver/main.go b/staging/src/k8s.io/sample-apiserver/main.go
index 708a2a43125..d6dedc9b989 100644
--- a/staging/src/k8s.io/sample-apiserver/main.go
+++ b/staging/src/k8s.io/sample-apiserver/main.go
@@ -17,16 +17,19 @@ limitations under the License.
 package main
 
 import (
+	"fmt"
 	"os"
 
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	"k8s.io/component-base/cli"
+	"k8s.io/kubernetes/pkg/util/env"
 	"k8s.io/sample-apiserver/pkg/cmd/server"
 )
 
 func main() {
 	stopCh := genericapiserver.SetupSignalHandler()
 	options := server.NewWardleServerOptions(os.Stdout, os.Stderr)
+	fmt.Println(env.GetEnvAsStringOrFallback("HACKHACKHACK", "boo"))
 	cmd := server.NewCommandStartWardleServer(options, stopCh)
 	code := cli.Run(cmd)
 	os.Exit(code)
```

...and ran update-vendor and got:

```
+++ [1208 08:48:49] logfile at /tmp/update-vendor.VsD1/update-vendor.log
+++ [1208 08:48:49] go.mod: update staging references
+++ [1208 08:49:01] go.mod: propagate to staging modules
+++ [1208 08:49:03] go.mod: sorting staging modules
+++ [1208 08:49:50] go.mod: tidying
++ date '+[%m%d %H:%M:%S]'
+ timestamp='[1208 08:50:16]'
+ echo '!!! [1208 08:50:16] 1 disallowed k8s.io/sample-apiserver -> k8s.io/kubernetes dependencies exist via the following imports: # k8s.io/kubernetes/pkg/util/env
k8s.io/sample-apiserver
k8s.io/kubernetes/pkg/util/env'
!!! [1208 08:50:16] 1 disallowed k8s.io/sample-apiserver -> k8s.io/kubernetes dependencies exist via the following imports: # k8s.io/kubernetes/pkg/util/env
k8s.io/sample-apiserver
k8s.io/kubernetes/pkg/util/env
+ shift
An error has occurred. Please see more details in /tmp/update-vendor.VsD1/update-vendor.log
```

Incidentally< i wanted to fix the dat format but I don't know if any tooling tries to parse it :)